### PR TITLE
Added logs and did some nits on the code

### PR DESCRIPTION
**Pull request by vishnuchalla**
### Description
Added logs and few nits in the code.

### Testing
Full log here: http://pastebin.test.redhat.com/1110069

### CODE REVIEW

File: cmd/clair-load-test/reports.go
Comment (hdonnay) [inline cmd/clair-load-test/reports.go]: I don't understand why this is here -- it's a real code smell to unconditionally sleep for no reason.

File: cmd/clair-load-test/reports.go
Comment (hdonnay) [inline cmd/clair-load-test/reports.go]: Why does this need a deep copy? That seems to indicate this isn't actually a configuration and is modified.

**Review by hdonnay — CHANGES_REQUESTED** [review]
I don't understand why a warmup is needed. If there's unavailability of layers, that's not Clair's issue and the setup process should wait for the storage to become available.

File: cmd/clair-load-test/reports.go
Comment (vishnuchalla) [reply cmd/clair-load-test/reports.go]: Sorry about that. Was doing some testing and forgot to remove it.

File: cmd/clair-load-test/reports.go
Comment (vishnuchalla) [reply cmd/clair-load-test/reports.go]: In the warmup step, we did not want to index the results to ES. So did a clone of existing config and setup some dummy params so that we can just use it for warmup and not store any of those results.

File: cmd/clair-load-test/reports.go
Comment (vishnuchalla) [inline cmd/clair-load-test/reports.go:218]: Keeping only this timeout, because we observed index report retrying requests which is affecting the other API endpoint results which are dependent on index report POST operation. Adding this timeout will provide POST index_report endpoint with enough time to run requests and retry them before we move on to the subsequent endpoints that are dependent on this. Let us know if this doesn't make any sense. But we saw improvements in the tests results when we did this.

**Comment by vishnuchalla** [comment]
@crozzy Can I get a review on this as well. Its just a few logs and nits in the code as QE-team's ask.

File: attacker/attacker.go
Comment (crozzy) [inline attacker/attacker.go]: Only handle errors once if you can,  so I'd say don't log here with no context, leave it to the calling function to log the error (as it does now).

File: attacker/attacker.go
Comment (crozzy) [inline attacker/attacker.go]: Same here

File: cmd/clair-load-test/reports.go
Comment (crozzy) [inline cmd/clair-load-test/reports.go]: I'd use `Duration`'s `String()` method, i.e. something like:
```go
zlog.Info(ctx).Stringer("duration", elapsedTime).Msg("Total time taken for completion")
// 1h3m5s
```

File: attacker/apis.go
Comment (crozzy) [inline attacker/apis.go:15]: A lot of these seem like genuine `Debug()` messages, if QE wants to see this output, can they just run with log level `debug`?

**Review by crozzy — COMMENTED** [review]
I added a couple of comments, the commits also need to be squashed before merging and the commit message no longer applies to this commit so it'd be great if you could amend that. Thanks

File: attacker/apis.go
Comment (vishnuchalla) [reply attacker/apis.go:15]: I changed it to `Info` as I felt like there are no enough logs when enabled `Info` as log level. I am thinking to keep these as `Info` and enable `Debug` only in case of error messages. Does that sound okay?

File: attacker/attacker.go
Comment (vishnuchalla) [reply attacker/attacker.go]: Keeping log level as `Debug` for these type of errors.

File: attacker/attacker.go
Comment (vishnuchalla) [reply attacker/attacker.go]: Changing this log level to Debug, as its an error case.

File: cmd/clair-load-test/reports.go
Comment (vishnuchalla) [reply cmd/clair-load-test/reports.go]: Changed as suggested.

File: attacker/attacker.go
Comment (crozzy) [inline attacker/attacker.go]: This is overly complicated:
```go
if err != nil {
    return err
}
zlog.Info(ctx).Msg(resp)
```

File: attacker/attacker.go
Comment (crozzy) [reply attacker/attacker.go]: IMO the log line shouldn't exist, if you want to add more information just format the error message as you do in other places, the error eventually gets logged and it makes a lot more sense to the user. But up to you, it's not part of this PR anymore.

**Review by crozzy — CHANGES_REQUESTED** [review]
This still needs the commits sorted out, I'm not going to block it based on log levels

File: attacker/attacker.go
Comment (vishnuchalla) [reply attacker/attacker.go]: Changed as suggested

**Comment by vishnuchalla** [comment]
> This still needs the commits sorted out, I'm not going to block it based on log levels

Updated code with suggestions and cleaned up the commits.

File: cmd/clair-load-test/reports.go
Comment (crozzy) [inline cmd/clair-load-test/reports.go]: This is already reviewed and merged, not sure why it's back again

File: attacker/attacker.go
Comment (crozzy) [inline attacker/attacker.go]: What is the point in the `else` here?

**Review by crozzy — CHANGES_REQUESTED** [review]
I think something got messed up in the squashing, there are a mix of changes in there now

File: attacker/attacker.go
Comment (vishnuchalla) [reply attacker/attacker.go]: It logs this information on what action was performed by indexer while indexing docs to ES.
```
2023-10-23T19:00:37Z INF Indexing finished in 9ms: created=1
```

File: cmd/clair-load-test/reports.go
Comment (vishnuchalla) [reply cmd/clair-load-test/reports.go]: Updated the code.

**Comment by vishnuchalla** [comment]
> I think something got messed up in the squashing, there are a mix of changes in there now

Fixed it.

File: attacker/attacker.go
Comment (crozzy) [reply attacker/attacker.go]: Great, but as I originally suggested you don't need the `else` I don't think, just log after the `return err`

**Review by crozzy — APPROVED** [review]

**Pull request merged**
